src/filemenu/filemenu_createfile.c

```diff
@@ -0,0 +1,485 @@
+            set_window_update(WINDOW_ID_FILEMENU_MONO, (s32) &filemenu_update_show_with_rotation);
+            set_window_update(WINDOW_ID_FILEMENU_OPTION_LEFT, (s32) &filemenu_update_show_with_rotation);
+            set_window_update(WINDOW_ID_FILEMENU_OPTION_CENTER, (s32) &filemenu_update_show_with_rotation);
+            set_window_update(WINDOW_ID_FILEMENU_OPTION_RIGHT, (s32) &filemenu_update_show_with_rotation);
+            set_window_update(WINDOW_ID_FILEMENU_OPTION_LEFT, (s32) &filemenu_update_show_with_rotation);
+            set_window_update(WINDOW_ID_FILEMENU_FILE0_INFO, (s32) &filemenu_update_show_with_rotation);
+            set_window_update(WINDOW_ID_FILEMENU_FILE1_INFO, (s32) &filemenu_update_show_with_rotation);
+            set_window_update(WINDOW_ID_FILEMENU_FILE2_INFO, (s32) &filemenu_update_show_with_rotation);
+            set_window_update(WINDOW_ID_FILEMENU_FILE3_INFO, (s32) &filemenu_update_show_with_rotation);
+            filemenu_currentMenu = 0;
+            return;
+        }
+
+        for (i = filemenu_filename_pos; i < ARRAY_COUNT(filemenu_filename); i++) {
+            filemenu_filename[i] = 0xF7;
+        }
+    }
+
+    if (filemenu_pressedButtons & BUTTON_START) {
+        for (i = 0; i < ARRAY_COUNT(filemenu_filename); i++) {
+            if (filemenu_filename[i] != 0xF7) {
+                break;
+            }
+        }
+        if (i == ARRAY_COUNT(filemenu_filename)) {
+            sfx_play_sound(SOUND_MENU_ERROR);
+            return;
+
+        }
+
+        sfx_play_sound(SOUND_MENU_NEXT);
+        set_window_update(WINDOW_ID_FILEMENU_YESNO_OPTIONS, (s32) filemenu_update_show_name_confirm);
+
+        gWindows[WINDOW_ID_FILEMENU_YESNO_OPTIONS].pos.y = 121;
+        gWindows[WINDOW_ID_FILEMENU_YESNO_OPTIONS].width = 69;
+        gWindows[WINDOW_ID_FILEMENU_YESNO_OPTIONS].height = 44;
+        gWindows[WINDOW_ID_FILEMENU_YESNO_OPTIONS].pos.x = ((gWindows[WINDOW_ID_FILEMENU_YESNO_OPTIONS].parent != -1)
+                            ? (gWindows[gWindows[WINDOW_ID_FILEMENU_YESNO_OPTIONS].parent].width / 2)
+                            : SCREEN_WIDTH / 2) - gWindows[WINDOW_ID_FILEMENU_YESNO_OPTIONS].width / 2;
+
+        gWindows[WINDOW_ID_FILEMENU_YESNO_PROMPT].pos.y = -70;
+        gWindows[WINDOW_ID_FILEMENU_YESNO_PROMPT].width = 164;
+        gWindows[WINDOW_ID_FILEMENU_YESNO_PROMPT].height = 62;
+        gWindows[WINDOW_ID_FILEMENU_YESNO_PROMPT].pos.x = ((gWindows[WINDOW_ID_FILEMENU_YESNO_PROMPT].parent != -1)
+                            ? (gWindows[gWindows[WINDOW_ID_FILEMENU_YESNO_PROMPT].parent].width / 2)
+                            : SCREEN_WIDTH / 2) - gWindows[WINDOW_ID_FILEMENU_YESNO_PROMPT].width / 2;
+
+        newMenu2 = filemenu_menus[1];
+        filemenu_currentMenu = 1;
+        newMenu2->page = 2;
+        filemenu_set_selected(newMenu2, 0, 0);
+    }
+}
+
+static const f32 padding[2] = { 0.0f, 0.0f }; // not sure why this is needed
```

Comment:
File split?

Reply:
If I remove this line, rodata section doesn't change at all, but `filemenu_RODATA_END` becomes wrong, which is referenced in `state_file_select.c`. So I'm not sure if splitting can fix this.